src/components/ResourcePicker/tests/ResourcePicker.test.tsx

```diff
@@ -68,6 +68,27 @@ describe('<ResourcePicker />', () => {
       expect(onSelection).toHaveBeenCalledWith(selectionPayload);
     });
 
+    it('calls the new selection callback when changed', () => {
+      const onSelection = jest.fn();
+      const selectionPayload = {selection: []};
+      const {resourcePicker} = mountWithAppBridge(
+        <ResourcePicker resourceType="Product" open />,
+      );
+
+      resourcePicker.setProps({onSelection});
+
+      expect(appBridgeResourcePickerMock.subscribe).toHaveBeenCalledTimes(1);
+
```

Comment:
Let’s also check that unsubscribe was called once `expect(appBridgeResourcePickerMock.unsubscribe).toHaveBeenCalledTimes(1);`